qiskit/algorithms/linear_solvers/numpy_linear_solver.py

```diff
@@ -0,0 +1,77 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""The Numpy LinearSolver algorithm (classical)."""
+
+from typing import List, Union, Optional, Callable
+import logging
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .linear_solver import LinearSolverResult, LinearSolver
+from .observables.linear_system_observable import LinearSystemObservable
+
+logger = logging.getLogger(__name__)
+
+
+class NumPyLinearSolver(LinearSolver):
+    """
+    The Numpy LinearSolver algorithm (classical).
+
+    This linear system solver computes the exact value of the given observable(s) or the full
+    solution vector if no observable is specified.
+    """
+    def solve(self, matrix: Union[np.ndarray, QuantumCircuit],
+              vector: Union[np.ndarray, QuantumCircuit],
+              observable: Optional[Union[LinearSystemObservable, BaseOperator,
+                                         List[BaseOperator]]] = None,
+              post_rotation: Optional[Union[QuantumCircuit, List[QuantumCircuit]]] = None,
+              post_processing: Optional[Callable[[Union[float, List[float]]],
+                                                 Union[float, List[float]]]] = None) \
+            -> LinearSolverResult:
+        """
+        solve the system and compute the observable(s)
+
+        Args:
+            matrix: The matrix specifying the system, i.e. A in Ax=b.
+            vector: The vector specifying the right hand side of the equation in Ax=b.
+            observable: Information to be extracted from the solution.
+                Default is `EuclideanNorm`
+            post_rotation: Circuit to be applied to the solution to extract information.
+            post_processing: Function to compute the value of the observable.
```

Comment:
Since these are are optional maybe say 'Optional circuit to be applied....' maybe they have no default - if there is some default post_rotation/processing done, if this is None, can we state it here.

Reply:
changed the description